### PR TITLE
feat: add support for custom topology label

### PR DESCRIPTION
**Pull request by varnastadeus**

### CODE REVIEW

File: cmd/kg/main.go [hunk 1]
```diff
@@ -92,6 +92,7 @@ func Main() error {
 	local := flag.Bool("local", true, "Should Kilo manage routes within a location?")
 	logLevel := flag.String("log-level", logLevelInfo, fmt.Sprintf("Log level to use. Possible values: %s", availableLogLevels))
 	master := flag.String("master", "", "The address of the Kubernetes API server (overrides any value in kubeconfig).")
+	topologyLabel := flag.String("topology-label", k8s.RegionLabelKey, "Kubernetes node label used to group logical nodes.")
```
Comment (squat) [inline cmd/kg/main.go:95]: How about:
```suggestion
	topologyLabel := flag.String("topology-label", k8s.RegionLabelKey, "Kubernetes node label used to group nodes into logical locations.")
```

File: cmd/kgctl/main.go [hunk 2]
```diff
@@ -110,6 +111,7 @@ func main() {
 	cmd.PersistentFlags().StringVar(&granularity, "mesh-granularity", string(mesh.LogicalGranularity), fmt.Sprintf("The granularity of the network mesh to create. Possible values: %s", availableGranularities))
 	cmd.PersistentFlags().StringVar(&kubeconfig, "kubeconfig", os.Getenv("KUBECONFIG"), "Path to kubeconfig.")
 	cmd.PersistentFlags().Uint32Var(&opts.port, "port", mesh.DefaultKiloPort, "The WireGuard port over which the nodes communicate.")
+	cmd.PersistentFlags().StringVar(&topologyLabel, "topology-label", k8s.RegionLabelKey, "Kubernetes node label used to group logical nodes.")
```
Comment (squat) [inline cmd/kgctl/main.go:114]: Same here

File: pkg/k8s/backend.go [hunk 3]
```diff
@@ -59,8 +59,8 @@ const (
 	locationAnnotationKey        = "kilo.squat.ai/location"
 	persistentKeepaliveKey       = "kilo.squat.ai/persistent-keepalive"
 	wireGuardIPAnnotationKey     = "kilo.squat.ai/wireguard-ip"
-
-	regionLabelKey  = "topology.kubernetes.io/region"
+	// RegionLabelKey is the key for region label.
```
Comment (squat) [inline pkg/k8s/backend.go:62]:
```suggestion
	// RegionLabelKey is the key for the well-known Kubernetes topology region label.
```

**Comment by squat** [comment]
@varnastadeus thanks for contributing this. The code looks great. I just have some comments around text.

Could you also add a small description summarizing how to use this feature in the https://github.com/squat/kilo/blob/master/docs/topology.md doc?

:rocket:

File: cmd/kgctl/main.go (hunk 2 above)
Comment (varnastadeus) [inline cmd/kgctl/main.go:114]:
```suggestion
	cmd.PersistentFlags().StringVar(&topologyLabel, "topology-label", k8s.RegionLabelKey, "Kubernetes node label used to group nodes into logical locations.")
```

**Comment by varnastadeus** [comment]
> @varnastadeus thanks for contributing this. The code looks great. I just have some comments around text.
> 
> Could you also add a small description summarizing how to use this feature in the https://github.com/squat/kilo/blob/master/docs/topology.md doc?
> 
> 🚀

@squat did the changes requested. Not sure why container building fails, do you have any ideas?

**Comment by squat** [comment]
The code looks good! Tests pass but total CI fails; it's a problem with the new Docker rate limits :///

```
toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

I think this will be fixed once we move to github actions, as I believe GitHub has negotiated a special agreement with Docker.

File: cmd/kg/main.go [hunk 4]
```diff
@@ -92,6 +92,7 @@ func Main() error {
 	local := flag.Bool("local", true, "Should Kilo manage routes within a location?")
 	logLevel := flag.String("log-level", logLevelInfo, fmt.Sprintf("Log level to use. Possible values: %s", availableLogLevels))
 	master := flag.String("master", "", "The address of the Kubernetes API server (overrides any value in kubeconfig).")
+	topologyLabel := flag.String("topology-label", k8s.RegionLabelKey, "Kubernetes node label used to group nodes into logical locations.")
```
Comment (squat) [inline cmd/kg/main.go:95]: It just occurred to me, should we be clear that this is the label _key_? Or is that not necessary?

Maybe we can look at what K8s calls this now on their well known labels page https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/
They make no mention of "key" only "label" and "value". So maybe this is fine as it. WDYT?

Comment (varnastadeus) [reply cmd/kg/main.go:95]: I think "label" is pretty clear :]

**Review by squat — APPROVED** [review]

**Pull request merged**